doc/rtd/topics/network-config-format-v1.rst

```diff
@@ -303,7 +319,7 @@ packet- or frame-based network.  Specifying ``mtu`` is optional.
        # Physical interfaces.
        - type: physical
          name: eth0
-         mac_address: "c0:d6:9f:2c:e8:80"
+         mac_address: c0:d6:9f:2c:e8:80
        # VLAN interface.
```

Comment:
i think we should just be consistent with the quoting.

Reply:
This PR is attempting to be consistent, that is "quote only when necessary and when quoting only use single quotes".

Reply:
i understand, but i feel that's just going to create a bit more confusion than necessary, unlike consistent quoting

Reply:
The problem is that quoting **all** MAC address in the testcases will cause some of the testcases to fail - in some testcases the code converts YAML into Python datastructures and back to YAML which ends up with MAC addresses without quotes (where not needed) and so testcases fail where the expected output and generated output do not match. This is the YAML libraries doing what they're designed to do - only quote strings when needed.

Reply:
> i understand, but i feel that's just going to create a bit more confusion than necessary, unlike consistent quoting

The issue is that tooling that generates valid (1.1) YAML will not quote things in the manner you are describing as consistent: it will have this mix of quoted and non-quoted values, conditional on whether they would be interpreted as an integer if they weren't quoted. So if someone looks at the output that their tool generates, and see unquoted MAC addresses, they may correctly notice that their tooling is inconsistent with our docs and so _incorrectly_ infer that their tooling is doing something wrong. (Or, put another way, quoting all MAC addresses in our docs is _not_ consistent with what the YAML spec requires.)

So: I can see your point and I agree that this _is_ inconsistent, but I think it's the YAML 1.1 spec that's leading to this inconsistency: inconsistency in the specification of our input format is going to be reflected in inconsistency in our documentation of inputs in that format. As a result, I don't think there is one clearly correct answer to this and as Dermot is the one working on this, I'm happy to take their lead.